### PR TITLE
VSP-1467: Paysafe service charge amount is passed to Noire

### PR DESCRIPTION
**Pull request by ybenzine**
Paysafe service charge is passed to Noire to be reused by the invoice and order logic from the response object:
- Added `setServiceCharge` and `getServiceCharge` to `AuthorizeRequest`
- Added `customParameters[service_charge]` as one of the parameters passed to Noire (via the `getData` method)
- Added `getServiceCharge` to Paysafe's `CompleteAuthorizeResponse` class

### CODE REVIEW

**Comment by ybenzine** [comment]
Right, I understand that! I felt it was easier to bundle it than to create its own ticket. I'll add the unit tests now.

File: src/Message/Paysafecard/AuthorizeRequest.php [hunk 1]
```diff
@@ -45,7 +65,8 @@ public function getData()
             'paymentBrand' => $this->getBrand(),
             'paymentType' => $this->getPaymentType(),
             'shopperResultUrl' => $this->getShopperResultUrl(),
-            'merchantTransactionId' => $this->getTransactionId()
+            'merchantTransactionId' => $this->getTransactionId(),
+            'customParameters[service_charge]' => $this->getServiceCharge(),
```
Comment (angelxmoreno) [inline src/Message/Paysafecard/AuthorizeRequest.php]: lol, wtf is this? they want a string with braces as a key to an array ?
but they can't handle a fu****g domain with underscores? ROFL.

**Comment by ybenzine** [comment]
retest this please

**Comment by ybenzine** [comment]
retest sonar please

**Comment by ybenzine** [comment]
run sonar

**Comment by ybenzine** [comment]
retest this please

**Pull request merged**